File: Python/001_Lista_intro/Ex_007.py
"""
7) Fazer um algoritmo para resolver uma equação do primeiro grau.
"""
import math
import numpy as np #biblioteca para fazer direto equacao de segundo grau
print("Digite os valores para: a, b e c")

a = float(input())
b = float(input())
c = float(input())

raizes = np.roots([a, b, c])
print("As raizes da equacao sao:", raizes)

File: XDES01 - Linguagem C/001_Lista_intro/Ex_007.c
//7) Fazer um algoritmo para resolver uma equação do primeiro grau
#include <stdio.h>
int main ()
{
    int a, b, x;
    
    scanf("%d %d", &a, &b);
    
    if (a == 0) {
        if (b == 0) {
            printf("A equacao possui infinitas solucoes.\n");
        } else {
            printf("A equacao nao possui solucao.\n");
        }
    } else {
        x = -b / a;
        printf("A solucao da equacao e: x = %.2f\n", x);
    }

    return 0;
}